docs/entities/renderer.md

```diff
@@ -0,0 +1,584 @@
+
+`RenderLayer`s define a `#render` method, which - surprise! - renders the layer. As with most other render methods, you can basically render whatever you want in here. However, a very common use case is to render a separate model in here, for example for armor or similar pieces of equipment.
+
+For this, we first need a model we can render. We use the `EntityModel` class to do this. `EntityModel`s are basically a list of cubes and associated textures for the renderer to use. They are commonly created statically when the entity renderer's constructor is first created.
+
+:::note
+Since we now operate on `LivingEntityRenderer`s, the following code will assume that `MyEntity extends LivingEntity` and `MyEntityRenderState extends LivingEntityRenderState`, to match generic type bounds.
+:::
+
+### Creating an Entity Model Class and a Layer Definition
+
+Let's start by creating an entity model class:
+
+```java
+public class MyEntityModel extends EntityModel<MyEntityRenderState> {}
+```
+
+Note that in the above example, we directly extend `EntityModel`; depending on your use case, it might be more appropriate to use one of the subclasses instead. When creating a new model, it is recommended you have a look at whatever existing model is closest to your use case, and then work from there.
+
+Next, we create a `LayerDefinition`. A `LayerDefinition` is basically a list of cubes that we can then bake to an `EntityModel`. Defining a `LayerDefinition` looks something like this:
+
+```java
+public class MyEntityModel extends EntityModel<MyEntityRenderState> {
+    // A static method in which we create our layer definition. createBodyLayer() is the name
+    // most vanilla models use. If you have multiple layers, you will have multiple of these static methods.
+    public static LayerDefinition createBodyLayer() {
+        // Create our mesh.
+        MeshDefinition mesh = new MeshDefinition();
+        // The mesh initially contains no object other than the root, which is invisible (has a size of 0x0x0).
+        PartDefinition root = mesh.getRoot();
+        // We add a head part.
+        PartDefinition head = root.addOrReplaceChild(
+            // The name of the part.
+            "head",
+            // The CubeListBuilder we want to add. While it is possible to add multiple cubes into one part,
+            // it is generally discouraged and only one cube per PartDefinition should be used.
+            CubeListBuilder.create()
+                // The UV coordinates to use within the texture. Texture binding itself is explained below.
+                // In this example, we start at U=10, V=20.
+                .texOffs(10, 20)
+                // Add our cube. Again, while multiple can be added, it is recommended to only add one.
+                .addBox(
+                    // The origin of the cube, relative to the parent object's position.
```

Comment:
Sort of? The 'origin' is a bit weird. For a root part definition, it's relative to the `PartPose` offset, which is in itself relative to the entity's position. You should notice that the y-direction is flipped here where up subtracts and down adds. For a sub-part, however, it is also relative to its `PartPose` offset, which is relative to the parent part definition, and down the chain.

It should also be mentioned that this applies to the top corner of the cube, and not the center.